library/src/main/java/com/github/windsekirun/naraeimagepicker/activity/NaraePickerActivity.kt

```diff
@@ -43,12 +46,47 @@ class NaraePickerActivity : AppCompatActivity() {
         SelectedItem.setLimits(PickerSet.getSettingItem().pickLimit)
         requestFileViewMode = PickerSet.getSettingItem().viewMode == ViewMode.FileView
 
-        RPermission.instance.checkPermission(this, arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE)) { integer, _ ->
-            if (integer == RPermission.PERMISSION_GRANTED) {
+        checkPermission()
+    }
+
+    private fun checkPermission() {
+        val storageReadPermission = ContextCompat.checkSelfPermission(
+                this, Manifest.permission.READ_EXTERNAL_STORAGE
+        ) == PackageManager.PERMISSION_GRANTED
+        when {
+            storageReadPermission -> {
                 initFragment(if (requestFileViewMode) FragmentMode.All else FragmentMode.Album)
-            } else {
-                setResult(Activity.RESULT_CANCELED)
+            }
+            else -> when {
+                Build.VERSION.SDK_INT >= Build.VERSION_CODES.M -> {
+                    requestPermissions(
+                            arrayOf(
+                                    Manifest.permission.READ_EXTERNAL_STORAGE,
+                                    Manifest.permission.WRITE_EXTERNAL_STORAGE
+                            ), RC_READ_STORAGE
+                    )
+                }
+            }
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+            requestCode: Int,
+            permissions: Array<out String>,
+            grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        when (requestCode) {
+            RC_READ_STORAGE -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    initFragment(if (requestFileViewMode) FragmentMode.All else FragmentMode.Album)
+                } else {
+                    setResult(Activity.RESULT_CANCELED)
+                }
+                return
+            }
+            else -> {
```

Comment:
Since we don't need any permission except STORAGE, can be simplified by this statement.

```
if (requestCode != RC_READ_STORAGE) return

if (grantResults.firstOrNull() == PackageManager.PERMISSION_GRANTED) {
        initFragment(if (requestFileViewMode) FragmentMode.All else FragmentMode.Album)
} else {
        setResult(Activity.RESULT_CANCELED)
}
```